lib/cmds/update.ts

```diff
@@ -61,120 +61,138 @@ if (argv._[0] === 'update-run') {
+      } else {
+        outputDir = path.resolve(Config.getBaseDir(), options[Opt.OUT_DIR].getString());
       }
+      FileManager.makeOutputDirectory(outputDir);
+    }
+    let ignoreSSL = options[Opt.IGNORE_SSL].getBoolean();
+    let proxy = options[Opt.PROXY].getString();
+    let verbose = options[Opt.VERBOSE].getBoolean();
+
+    // setup versions for binaries
+    let binaries = FileManager.setupBinaries(options[Opt.ALTERNATE_CDN].getString());
+    binaries[StandAlone.id].versionCustom = options[Opt.VERSIONS_STANDALONE].getString();
+    binaries[ChromeDriver.id].versionCustom = options[Opt.VERSIONS_CHROME].getString();
+    if (options[Opt.VERSIONS_IE]) {
+      binaries[IEDriver.id].versionCustom = options[Opt.VERSIONS_IE].getString();
+    }
+    if (options[Opt.VERSIONS_GECKO]) {
+      binaries[GeckoDriver.id].versionCustom = options[Opt.VERSIONS_GECKO].getString();
+    }
+    binaries[AndroidSDK.id].versionCustom = options[Opt.VERSIONS_ANDROID].getString();
+    binaries[Appium.id].versionCustom = options[Opt.VERSIONS_APPIUM].getString();
+
+    // if the file has not been completely downloaded, download it
+    // else if the file has already been downloaded, unzip the file, rename it, and give it
+    // permissions
+    if (standalone) {
+      let binary = binaries[StandAlone.id];
+      updateBrowserFile(binary, outputDir);
+      promises.push(
+          FileManager.downloadFile(binary, outputDir, proxy, ignoreSSL)
+              .then((downloaded: boolean) => {
+                if (!downloaded) {
+                  logger.info(
+                      binary.name + ': file exists ' +
+                      path.resolve(outputDir, binary.filename(Config.osType(), Config.osArch())));
+                  logger.info(binary.name + ': ' + binary.versionCustom + ' up to date');
+                }
+              }));
+    }
+    if (chrome) {
+      let binary = binaries[ChromeDriver.id];
+      updateBrowserFile(binary, outputDir);
+      promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
+    }
+    if (gecko) {
+      let binary = binaries[GeckoDriver.id];
+      updateBrowserFile(binary, outputDir);
+      promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
+    }
+    if (ie) {
+      let binary = binaries[IEDriver.id];
+      binary.arch = Config.osArch();  // Win32 or x64
+      updateBrowserFile(binary, outputDir);
+      promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
+    }
+    if (ie32) {
+      let binary = binaries[IEDriver.id];
+      binary.arch = 'Win32';
+      updateBrowserFile(binary, outputDir);
+      promises.push(updateBinary(binary, outputDir, proxy, ignoreSSL));
+    }
+    if (android) {
+      let binary = binaries[AndroidSDK.id];
+      let sdk_path = path.resolve(outputDir, binary.executableFilename(Config.osType()));
+      let oldAVDList: string;
+
+      q.nfcall(fs.readFile, path.resolve(sdk_path, 'available_avds.json'))
```

Comment:
Not sure why this is part of the "hide outdated". Anyway, are q promises the same as es6 Promises?